pennylane_lightning/core/src/simulators/lightning_gpu/gates/tests/Test_StateVectorCudaManaged_Generators.cpp

```diff
@@ -1662,3 +1662,324 @@ TEMPLATE_TEST_CASE("StateVectorCudaManaged::applyGeneratorGlobalPhase",
+                                            wires, inverse, {-ep});
+                }
+            } else {
+                scale = gntr_sv.applyGenerator(
+                    controlled_gate_name, {control_wire_0, control_wire_1},
+                    {control_value_0, control_value_1}, wires, inverse);
+                if (inverse) {
+                    gate_svp.applyOperation(controlled_gate_name,
+                                            {control_wire_0, control_wire_1},
+                                            {control_value_0, control_value_1},
+                                            wires, inverse, {-ep});
+                    gate_svm.applyOperation(controlled_gate_name,
+                                            {control_wire_0, control_wire_1},
+                                            {control_value_0, control_value_1},
+                                            wires, inverse, {ep});
+                } else {
+                    gate_svp.applyOperation(controlled_gate_name,
+                                            {control_wire_0, control_wire_1},
+                                            {control_value_0, control_value_1},
+                                            wires, inverse, {ep});
+                    gate_svm.applyOperation(controlled_gate_name,
+                                            {control_wire_0, control_wire_1},
+                                            {control_value_0, control_value_1},
+                                            wires, inverse, {-ep});
+                }
+
+                auto result_gntr_sv = gntr_sv.getDataVector();
+                auto result_gate_svp = gate_svp.getDataVector();
+                auto result_gate_svm = gate_svm.getDataVector();
+
+                for (std::size_t j = 0; j < exp2(num_qubits); j++) {
+                    CHECK(-scale * imag(result_gntr_sv[j]) ==
+                          Approx(0.5 *
+                                 (real(result_gate_svp[j]) -
+                                  real(result_gate_svm[j])) /
+                                 ep)
+                              .margin(EP));
+                    CHECK(scale * real(result_gntr_sv[j]) ==
+                          Approx(0.5 *
+                                 (imag(result_gate_svp[j]) -
+                                  imag(result_gate_svm[j])) /
+                                 ep)
+                              .margin(EP));
+                }
+            }
+        }
+    }
+}
+
+TEMPLATE_TEST_CASE("Generators::applyControlledGenerator - 4 wires",
+                   "[GateGenerators]", float, double) {
+
+    const std::string controlled_gate_name =
+        GENERATE("DoubleExcitation", "DoubleExcitationMinus",
+                 "DoubleExcitationPlus", "GlobalPhase", "MultiRZ");
+    const std::size_t num_qubits = 6;
+    const TestType ep = 1e-3;
+    const TestType EP = 1e-4;
```

Comment:
Please review the variable names as I requested in a previous comment.

Reply:
fixed.